src/storage/cache/queue.rs

```diff
@@ -0,0 +1,203 @@
+use core::marker::PhantomData;
+use std::{
+    ptr::{null_mut, NonNull},
+    sync::atomic::{AtomicPtr, Ordering::*},
+};
+
+pub struct Queue<T> {
```

Comment:
This seems to be a reimplementation of `std::collections::LinkedList`. Is there any reason why you couldn't use `LinkedList`?

Reply:
Thanks for the suggestion @DelSkayn, I want to make this a lockfree queue, and wanted to experiment with lower level rust. I wasn't aware of the LinkedList, but will give it a go.

Reply:
I think there are lock-free queues in `crossbeam`, possibly including these:
- https://docs.rs/crossbeam/latest/crossbeam/queue/struct.SegQueue.html
- https://docs.rs/crossbeam/latest/crossbeam/deque/index.html

Reply:
Thanks for the suggestion @finnbear, currently I am not using this anywhere but as suggested, will look into this